<launch>

  <arg name="uav_name" default="$(optenv UAV_NAME uav1)"/>
  <arg name="debug" default="false" />
  <arg name="print_debug" default="true" />

  <arg unless="$(arg debug)" name="launch_prefix" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch"/>

  <group if="$(arg print_debug)">
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find balloon_planner)/config/rosconsole/debug.conf"/>
</group>
  <group ns="$(arg uav_name)">

    <node pkg="nodelet" type="nodelet" name="balloon_planner" args="standalone balloon_planner/BalloonPlanner" output="screen" launch-prefix="$(arg launch_prefix)" >

      <param name="uav_name" type="string" value="$(arg uav_name)" />
      <param name="uav_frame_id" type="string" value="$(arg uav_name)/fcu" />
      <param name="world_frame_id" type="string" value="$(arg uav_name)/local_origin" />
      <rosparam file="$(find balloon_planner)/config/plan_params.yaml" command="load" />

      <!-- Subscribers -->
      <remap from="~ball_filtered" to="balloon_filter/detection_used" />
      <remap from="~ball_prediction" to="balloon_filter/ball_prediction" />

      <!-- Publishers -->
      <!-- <remap from="~commanded_trajectory" to="~commanded_trajectory" /> -->
      <remap from="~commanded_trajectory" to="control_manager/mpc_tracker/set_trajectory" />
      <remap from="~debug_trajectory" to="~debug_trajectory" />

    </node>

  </group>

</launch>

